include/rm.h

```diff
@@ -192,20 +192,22 @@ enum door_states {
 /*
  * The four directions for a DrawBridge.
  */
-#define DB_NORTH 0
-#define DB_SOUTH 1
-#define DB_EAST 2
-#define DB_WEST 3
-#define DB_DIR 3 /* mask for direction */
+#define DB_NORTH 0x0
+#define DB_SOUTH 0x1
+#define DB_EAST  0x2
+#define DB_WEST  0x3
+#define DB_DIR   0x3 /* mask for direction */
 
 /*
  * What's under a drawbridge.
  */
-#define DB_MOAT 0
-#define DB_LAVA 4
-#define DB_ICE 8
-#define DB_FLOOR 16
-#define DB_UNDER 28 /* mask for underneath */
+#define DB_MOAT  0x0
+#define DB_LAVA  0x4
+#define DB_ICE   0x8
+#define DB_FLOOR 0xc
+#define DB_UNDER 0xc /* mask for underneath */
```

Comment:
Previously, `DB_UNDER = DB_LAVA | DB_ICE | DB_FLOOR`. In this change, `DB_UNDER = DB_FLOOR`. Does this not introduce bugs in code that uses DB_UNDER?

Reply:
It's still `DB_MOAT | DB_LAVA | DB_ICE | DB_FLOOR` (`0x0 | 0x4 | 0x8 | 0xc` is `0xc`), it's just that because they are mutually exclusive states, there can be overlap -- in other words, the checks are always `(flags & DB_UNDER) == DB_MOAT`, never `(flags & DB_MOAT) != 0`.  So it's not necessary to give each one a unique bit, as long as the values themselves are unique and distinguishable.

But this hack to save a bit is also not really necessary if you've expanded `flags`.